src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -6250,21 +6250,24 @@ private sealed class NamespaceCompletion : TypeCompletionBase
         {
             internal string Namespace;
 
-            internal override CompletionResult GetCompletionResult(string keyMatched, string prefix, string suffix)
+            internal override CompletionResult GetCompletionResult(string keyMatched, string prefix, string suffix, string namespaceToRemove)
             {
+                string completion = string.IsNullOrEmpty(namespaceToRemove)
+                    ? string.Create(CultureInfo.InvariantCulture, $"{prefix}{Namespace}{suffix}")
+                    : string.Create(CultureInfo.InvariantCulture, $"{prefix}{Namespace.AsSpan(namespaceToRemove.Length + 1)}{suffix}");
```

Comment:
@MartinGC94 Thanks for using explicit culture and new API! Please do the same in elsewhere of the PR.